src/Nuget.PackageIndex/SettingsJson.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Nuget.PackageIndex.Abstractions;
+
+namespace Nuget.PackageIndex
+{
+    /// <summary>
+    /// Represents settings.json file located in the index directory. If file does not
+    /// exist, creates a new settings.json with default settings. When user modifies, settings.json
+    /// changes are in effect after VS is re-opened.
+    /// </summary>
+    public class SettingsJson : ISettingsJson
+    {
+        private const string SettingJsonFileName = "settings.json";
+        private const string DefaultJsonContent = @"{
+    'includePackagePatterns' : []
+}
+";
+        /// <summary>
+        /// These are prefixes for package names that are shipped under,
+        /// %ProgramFiles(x86)%\Microsoft Web Tools\DNU. We do look for packages 
+        /// at multiple local folders (sources), but index only currated list of 
+        /// packages. If users want to add other packages to the index they could 
+        /// update file %UserProfile%\AppData\Local\Microsoft\PackageIndex\settings.json
+        /// and add a regex for their packages to "include" list.
+        /// </summary>
+        private readonly string[] _defaultIncludeRules = new []
+        {
+            @"dnx\..+",
+            @"entityframework\..+",
+            @"entityframework7\..+",
```

Comment:
Try to figure out how to track changes to package names.

Also did you consider supporting disabling default includes (didn't think it through, so I'm not sure if necessary or meaningful just asking the question)